cli/azd/cmd/container.go

```diff
@@ -209,17 +211,28 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		},
 	)
 
-	// Lazy loads the environment from the Azd Context when it becomes available
+	// Lazy loads an existing environment, erroring out if not available
+	// One can repeatedly call GetValue to wait until the environment is available.
 	container.RegisterSingleton(
 		func(lazyAzdContext *lazy.Lazy[*azdcontext.AzdContext], envFlags envFlag) *lazy.Lazy[*environment.Environment] {
 			return lazy.NewLazy(func() (*environment.Environment, error) {
-				_, err := lazyAzdContext.GetValue()
+				azdCtx, err := lazyAzdContext.GetValue()
 				if err != nil {
 					return nil, err
 				}
 
-				var env *environment.Environment
-				err = container.Resolve(&env)
+				environmentName := envFlags.environmentName
```

Comment:
@wbreza Do you feel better about this change? Hooks will end up not triggering the environment prompt.

However, I'm not sure if the `post-up` hook actually reloads the environment. I looked at the code and it seemed that it wouldn't.

Reply:
Looks like this is indeed an existing issue. Will create an issue for it.